apps/messaging-app-extn-playground/webhooks/whatsapp_handler.js

```diff
@@ -0,0 +1,58 @@
+const Router = require("express").Router;
+const router = new Router();
+const util = require("../api/util");
+const debug = util.debug;
+// [🔴 IMPORTANT]
+// For production-ready apps, please follow the instructions mentioned in the link below
+// https://developers.facebook.com/docs/whatsapp/cloud-api/get-started#configure-webhooks
+// Similarly, WhatsApp has a retry logic. Make sure you respond with an appropriate status code
+
+
+// You need to verify webhook registration or else it will not be registered
+router.get("/whatsapp/messages/hook", async (req, res) => {
+    debug("WhatsApp hook verified. You should be able to receive incoming messages");
+    const challenge = req.query["hub.challenge"];
+    res.send(challenge);
+});
+
+// Parse the incoming message and send it to Pipedrive
+// NB! Beware of WhatsApp's retry logic - for now, the endpoint is configured to return *success* no matter what
+router.post("/whatsapp/messages/hook", async (req, res) => {
+    if (req.body.object) {
+        if (
+            req.body.entry &&
+            req.body.entry[0].changes &&
+            req.body.entry[0].changes[0] &&
+            req.body.entry[0].changes[0].value.messages &&
+            req.body.entry[0].changes[0].value.messages[0]
+        ) {
```

Comment:
Since Node 14 it's possible to use optional chaining, this whole block could be replaced with

```suggestion
        if (req.body.entry?[0]?.changes?[0]?.value?.messages?[0]) {
```

Reply:
Looks a lot cleaner. But for better comprehension for beginners, shall I still retain the old way? (Helps them relate to WhatsApp payload a lot easier)